src/serializer/ResidualHeapSerializer.js

```diff
@@ -450,7 +450,9 @@ export class ResidualHeapSerializer {
     // First, let's figure out from which function and generator scopes this value is referenced.
     let functionValues = [];
     let generators = [];
+    let scopeList = [];
```

Comment:
This scopeList seems to be unnecessary. Down where you need it, you should be able to just say `Array.from(scopes)`.